pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java

```diff
@@ -148,7 +148,8 @@ public void setup(Context context)
     Preconditions.checkState(_localSegmentDir.mkdir());
     Preconditions.checkState(_localSegmentTarDir.mkdir());
 
-    _fileSystem = FileSystem.get(context.getConfiguration());
+    // Note: we made an assumption that input file and output path are using same filesystem.
+    _fileSystem = FileSystem.get(_hdfsSegmentTarDir.toUri(), context.getConfiguration());
```

Comment:
If I'm not misunderstanding, this one will always be local, right? It's assigned with FileOutputFormat.getWorkOutputPath(context)

Reply:
This filesystem should be same as where we want to store the segment tars.
If we use s3a path as segment tar dir. Then this should be s3a filesystem.